### PR TITLE
nixos: Force check the filesystem before resizing

### PR DESCRIPTION
**Pull request by knedlsepp**
###### Motivation for this change
This should fix filesystems not being resized on boot (#26910)
I simply added the filesystem check as suggested by the error log of `resize2fs`. 
On the next boot everything was resized automatically.
I'm however not sure what other implications this change might have.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/system/boot/stage-1-init.sh [hunk 1]
```diff
@@ -301,6 +301,7 @@ mountFS() {
         *x-nixos.autoresize*)
             if [ "$fsType" = ext2 -o "$fsType" = ext3 -o "$fsType" = ext4 ]; then
```
Comment (0xABAB) [inline nixos/modules/system/boot/stage-1-init.sh]: This can better be written with a case statement.

**Review by 0xABAB — CHANGES_REQUESTED** [review]
Just the thing I mentioned, otherwise it seems fine.

File: nixos/modules/system/boot/stage-1-init.sh (hunk 1 above)
Comment (edolstra) [inline nixos/modules/system/boot/stage-1-init.sh]: This line is not part of this PR, so suggesting improvements to it is kind of off-topic for this PR.

**Review by 0xABAB — APPROVED** [review]

**Comment by domenkozar** [comment]
I'm going to backport this to 17.03
